src/kartograf/atom_mapper.py

```diff
@@ -855,9 +854,76 @@ def suggest_mapping_from_rdmols(
 
         return mapping
 
+    @staticmethod
+    def _split_protein_components_molecules(protein: ProteinComponent) -> list[ProteinComponent]:
+        """
+        Aims at splitting a protein component into different protein components based on the
+        connectivity of the molecules that compose it. Useful for mapping multimer components
+        or proteins with structural waters or similarly.
+
+        This returns a list of ``ProteinComponent`` objects with the name attribute indicating the starting
+        index in the original component.
+        """
+        from rdkit.Chem.rdmolops import GetMolFrags
+        rdmol = protein.to_rdkit()
+        index_tuples = []
+        fragments = GetMolFrags(rdmol, asMols=True, sanitizeFrags=True, fragsMolAtomMapping=index_tuples)
+        components = [ProteinComponent(fragment, name=f"frag_{index_tuple[0]}") for fragment, index_tuple in zip(fragments, index_tuples)]
```

Comment:
Does it make sense to return a component here instead of just an rdmol?

We call `to_rdkit()` later, and it feels not super useful until we create our mappings later.

Reply:
Good point we can just add the numbering as a prop on the rdkit mol.